kotlinx-coroutines-core/common/src/flow/internal/ChannelFlow.kt

```diff
@@ -27,6 +27,17 @@ public abstract class ChannelFlow<T>(
     // buffer capacity between upstream and downstream context
     @JvmField val capacity: Int
 ) : Flow<T> {
+
+    private var shouldDropWhileBusy: Boolean = false
+
+    internal open fun update(
```

Comment:
There's no need to have a separate 3-argument `update`. You can just 3rd parameter to a regular `update` for simplicity. It is an internal method, so you don't have to preserve it.

Reply:
The other update fun is not internal, it's public. Should I still coalesce these to a single method?

Reply:
It is public in an internal class. It can be safely updated (put intented).